osu.Game/Overlays/MusicController.cs

```diff
@@ -98,6 +107,19 @@ protected override void LoadComplete()
             mods.BindValueChanged(_ => ResetTrackAdjustments(), true);
         }
 
+        private void beatmapsChanged(IRealmCollection<BeatmapSetInfo> sender, ChangeSet? changes)
+        {
+            Playlist.Clear();
+            Playlist.AddRange(sender.Select(x => x.ToLive(realm)));
+        }
```

Comment:
Do you mind explaining to me _why this works_?

`MusicController` has this realm subscription callback. But then also `Playlist` externally manages this. Somehow the end result in practice seems to be correct, in that `Playlist` wins, but _what even guarantees this ordering of execution?_

I dunno this PR keeps changing shape for no discernible reason and it seems to oscillate between being tolerable and inscrutable. I might have to reassess the issue from scratch myself at this point because we're three or four review rounds deep and this is looking no closer to merge.

Reply:
Music Controller is created first, then NowPlayingOverlay. Music Controller first updates the playlist, and then if there is another class that filters for example, it updates the playlist to the filter. Making it relevant.